core/store/upgrade.go

```diff
@@ -41,6 +41,19 @@ func (s *StoreUpgrades) IsDeleted(key string) bool {
 	return false
 }
 
+// IsRenamed returns true if the given key should be renamed
+func (s *StoreUpgrades) IsRenamed(key string) bool {
+	if s == nil {
+		return false
+	}
+	for _, re := range s.Renamed {
+		if re.OldKey == key {
+			return true
+		}
+	}
+	return false
+}
+
```

Comment:
I was wondering where the StoreUpgrades are sanity checked to ensure valid values and no duplicates that may cause issues?
Also using the same keys in  add/delete/ rename should be prevented.